src/Tizen.Peripheral/Tizen.Peripheral/Gpio.cs

```diff
@@ -0,0 +1,233 @@
+                throw ExceptionFactory.CreateException(ret);
+
+            var gpio = new Gpio(handle, pinNumber);
+
+            switch (mode)
+            {
+                case Mode.Input:
+                    ret = NativeGpio.SetEdgeMode(gpio._handle, NativeGpio.EdgeType.Both);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.In);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    gpio.SetIntteruptedCallback();
+                    gpio._interruptedEventCallback(handle, ErrorCode.None, IntPtr.Zero);
+                    break;
+                case Mode.OutputInitiallyLow:
+                    ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutLow);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    break;
+                case Mode.OutputInitiallyHigh:
+                    ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutHigh);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    break;
+            }
+
+            return gpio;
+        }
+
+        private void SetIntteruptedCallback()
+        {
+            _interruptedEventCallback = OnInterrupted;
+            var ret = NativeGpio.SetInterruptedCb(_handle, _interruptedEventCallback, IntPtr.Zero);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        private void OnInterrupted(IntPtr handle, ErrorCode error, IntPtr data)
+        {
+            PinUpdated?.Invoke(this, new PinUpdatedEventArgs(this, PinNumber, PinValue));
+        }
+
+        /// <summary>
+        /// Closes a GPIO pin.
+        /// </summary>
+        public void Close()
+        {
+            var ret = NativeGpio.UnsetInterruptedCb(_handle);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            ret = NativeGpio.Close(_handle);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Disposes GPIO.
+        /// </summary>
+        public void Dispose()
+        {
+            Close();
+        }
+
+        /// <summary>
+        /// GPIO pin number.
+        /// </summary>
+        public int PinNumber { get; }
+
+        /// <summary>
+        /// Sets or gets pin value.
+        /// </summary>
+        public Value PinValue
+        {
+            get
+            {
+                var ret = NativeGpio.Read(_handle, out uint value);
+                if (ret != ErrorCode.None)
+                    throw ExceptionFactory.CreateException(ret);
```

Comment:
Getter of `PinValue` creates an exception of type Exception, an exception type that should not be raised in a property getter. If this exception instance might be raised, convert this property into a method, or change this property's logic so that it no longer raises an exception.